### PR TITLE
Properly report Callable bound arguments

### LINKED ISSUE #63213: Emitted signal with parameters unbound has incorrect error for unexpected argument count.
### Godot version

v4.0.alpha.custom_build [d6b1dd485]
(Still reproducible following merge of dae5edf41c2287dea794c6a3a9d686479e71f10b)

### System information

Arch on 5.18.11-arch1-1

### Issue description

"Method expected 1 arguments, but called with 1.."
Can the extra `.` be fixed also?

```gdscript
extends Node

func _on_scene_child_entered_tree(_extraneous_parameter_unbound) -> void:
	print("hello")
```

```
E 0:00:04:0869   emit_signalp: Error calling from signal 'child_entered_tree' to callable: 'Node(scene.gd)::_on_scene_child_entered_tree': Method expected 1 arguments, but called with 1..
  <C++ Source>   core/object/object.cpp:1025 @ emit_signalp()
```

### Steps to reproduce

1) Bind signal child_entered_tree with 1 argument unbound.
2) Add an extraneous parameter to the callback.

### Minimal reproduction project

[test15.zip](https://github.com/godotengine/godot/files/9144311/test15.zip)

### PR DESCRIPTION
**Pull request by reduz**
Fixes #63213
Adds a function: Callable::get_bound_arguments_count() to query this in callables. Exposed to the engine API.

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]
Looks pretty good.

**Comment by kleonc** [comment]
> Adds a function: Callable::get_arguments_bound() to query this in callables.

I dislike the naming, I'd expect such named method to return the actual arguments bound, not their count.

>Exposed to the engine API.

Not sure if this is useful / makes sense given Callable doesn''t provide similar methods which would allow to:
- get argument count,
- get default argument count/values,
- get bound argument values.

**Comment by reduz** [comment]
@kleonc I can change the name, but this is useful if you are working with GDExtension in order to properly display errors from Callable::CallError, so it needs to be exposed.

**Comment by kleonc** [comment]
> @kleonc I can change the name

@reduz Now `get_amount_of_arguments_bound` is probably too wordy and I think not grammatically correct, AFAICT "amount" is for uncountable things. So I suggest `get_arguments_bound_count` instead. Should be consistent with the rest of the codebase, there are already methods like `get_child_count`, `get_point_count` etc.

>but this is useful if you are working with GDExtension in order to properly display errors from Callable::CallError, so it needs to be exposed.

Sure, absolutely fine. Just saying I won't be surprised when people will request other argument related info to be also obtainable from a Callable.

**Comment by reduz** [comment]
@kleonc Renamed to "get_bound_arguments_count", guess this should be good enough.

**Review by kleonc — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
